api/v3/Geometry.php

```diff
@@ -437,6 +437,42 @@ function _civicrm_api3_geometry_unarchive_spec(&$spec) {
   $spec['id']['type'] = CRM_Utils_Type::T_INT;
 }
 
+/**
+ * Geomety.getintersetion
+ *
+ * @param array $params
+ * @return array API result descriptor
+ * @throws API_Exception
+ */
+function civicrm_api3_geometry_getintersection($params) {
+  $result = CRM_CiviGeometry_BAO_Geometry::getGeometryIntesection($params);
+  return civicrm_api3_create_success($result, $params);
+}
+
+/**
+ * Geometry.getintersection API specification (optional)
+ * This is used for documentation and validation.
+ *
+ * @param array $spec description of fields supported by this API call
+ * @return void
+ * @see http://wiki.civicrm.org/confluence/display/CRMDOC/API+Architecture+Standards
+ */
+function _civicrm_api3_geometry_getintersection_spec(&$spec) {
+  $spec['geometry_a'] = [
+    'title' => E::ts('Geometry A'),
+    'api.required' => 1,
```

Comment:
@seamuslee001 do we need this `api.required` value for geometry B as well?

Reply:
@johntwyman i would argue no lets say you want to find all the State electorates that ward is in, you can specify the ward as geometry_a and then the ward collection as collection-id with no specific geometry_b

Reply:
Ah ok. That's a use case that hadn't occurred to me. As long as you pass either a geometry or a collection, then you can proceed. Makes sense. 👍

Reply:
@johntwyman that has now been implemented in new Line 448 of this file